scipy/stats/tests/test_stats.py

```diff
@@ -4737,6 +4737,208 @@ def test_std_zero(self):
         assert_equal(stats.sigmaclip(x)[0], x)
 
 
+class TestAlexanderGovern(object):
+    def test_compare_dtypes(self):
+        args = [[13, 13, 13, 13, 13, 13, 13, 12, 12],
+                [14, 13, 12, 12, 12, 12, 12, 11, 11],
+                [14, 14, 13, 13, 13, 13, 13, 12, 12],
+                [15, 14, 13, 13, 13, 12, 12, 12, 11]]
+        args_int16 = np.asarray(args, dtype=np.int16)
+        args_int32 = np.asarray(args, dtype=np.int32)
+        args_uint8 = np.asarray(args, dtype=np.uint8)
+        args_float64 = np.asarray(args, dtype=np.float64)
+
+        res_int16 = stats.alexandergovern(*args_int16)
+        res_int32 = stats.alexandergovern(*args_int32)
+        res_unit8 = stats.alexandergovern(*args_uint8)
+        res_float64 = stats.alexandergovern(*args_float64)
+
+        assert (res_int16.pvalue == res_int32.pvalue ==
+                res_unit8.pvalue == res_float64.pvalue)
+        assert (res_int16.statistic == res_int32.statistic ==
+                res_unit8.statistic == res_float64.statistic)
+
+    def test_bad_inputs(self):
+        # test for zero length input args
+        with assert_raises(ValueError) as e:
+            stats.alexandergovern([1, 2], [])
+        assert "cannot be zero-dimensional" in str(e)
```

Comment:
Also, an empty array is not zero dimensional, it is of zero _size_. This should be a different error message.

Reply:
Corrected to:

`size == 0 `-> "cannot be zero size"

`size == 1 `-> "cannot be zero dimensional"